doc/iterations/2023.6/shapes/addPagesToForm.adoc

```diff
@@ -0,0 +1,79 @@
+= (M) Add page concept in form view DSL
+
+== Problem
+
+It's not possible in the form to organise groups of widgets by pages.
+
+== Key Result
+
+The solution must offer the user the possibility to add or remove pages to a form description and add or remove groups of widgets to pages.
+
+== Solution
+
+Add to the view DSL the page _EClass_ _PageDescription_ with 4 attributes :
+
+. Name
+. Label Expression
+. Semantic Candidates Expression
+. Precondition Expression
+
+and 2 references :
+
+. `GroupDescription`
+. `ToolBarAction`
+
+In the new view model a `FormDescription` has now a reference to `PageDescription` and no `GroupDescription` anymore.
+Create the graphical representation in order to manage the pages of a form in the WYSIWYG editor.
+
+Each page can have several toolbar actions.
+
+=== Scenario 1—Manage from the explorer view
+
+User creates a new form description in a view with the explorer.
+A form description is initialized with a default page.
```

Comment:
I think the default Page here also have a default empty Group (but no ToolbarAction)

Reply:
OK, I will change this part. Does it mean that each new page should have a default empty group ?

Reply:
A page without a group, while possible, would not make much sense. Not creating a default one means in 99.9% of cases the user will have to create one, which involves multiple clicks. So I think that yes when we create a page it should have an empty group automatically created.

It's different for a `Group`, where we know the user will almost certainly create at least one widget in it, but we do not know which one. There is no such ambiguity on a `Page`, and in the rare cases where a user does not need the default Group we create, he can delete it in a single operation.

Note that this should not be done at the EMF factory level, but in the EMF.Edit part.
See `DiagramDescriptionItemProvider.collectNewChildDescriptors(Collection<Object>, Object)` for an example of how to customize this kind of behavior (and don't forget the `@generated NOT` annotation so that the customized code stays when the code is regenerated).